python/ray/air/tests/test_data_batch_conversion.py

```diff
@@ -12,44 +13,74 @@
 from ray.air.util.tensor_extensions.arrow import ArrowTensorArray
 
 
+@pytest.fixture(params=[True, False])
+def enable_automatic_tensor_extension_cast(request):
+    ctx = ray.data.context.DatasetContext.get_current()
+    original = ctx.enable_tensor_extension_casting
+    ctx.enable_tensor_extension_casting = request.param
+    yield request.param
+    ctx.enable_tensor_extension_casting = original
+
+
 def test_pandas_pandas():
     input_data = pd.DataFrame({"x": [1, 2, 3]})
     expected_output = input_data
     actual_output = convert_batch_type_to_pandas(input_data)
-    assert expected_output.equals(actual_output)
+    pd.testing.assert_frame_equal(expected_output, actual_output)
 
-    assert convert_pandas_to_batch_type(actual_output, type=DataType.PANDAS).equals(
-        input_data
+    actual_output = convert_pandas_to_batch_type(actual_output, type=DataType.PANDAS)
+    pd.testing.assert_frame_equal(actual_output, input_data)
+
+
+@pytest.mark.parametrize("use_tensor_extension", [True, False])
+def test_pandas_multi_dim_pandas(
+    enable_automatic_tensor_extension_cast, use_tensor_extension
```

Comment:
whats the difference between these 2 fixtures? can we add a comment explaining this?

Reply:
Sure I'll add a comment in the code; the difference is that `enable_automatic_tensor_extension_cast` is determining whether the automatic casting at the UDF boundaries happens, while `use_tensor_extension` determines whether the input to the batch conversion is using the tensor extension or not.

Reply:
Changed the fixture names to make it more clear, lmk if that suffices!